enhancements/profile-versioning.md

```diff
@@ -0,0 +1,245 @@
+
+1. Should the `version` attribute be immutable?
+2. Should a profile have multiple versions (e.g., PCI-DSS version 3.2.1 and 4.0.0)?
+3. What happens when a `Profile` extended by a `TailoredProfile` changes?
+4. Should the version string be incremented to include updates across operator releases (e.g., `1.4.0` compared to `1.4.0-1`)?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Upgrade / Downgrade Strategy
+
+The Compliance Operator currently supports upgrades, where fixes are rarely, if
+ever, backported. Due to this, the Compliance Operator does not support
+downgrades. The operator will upgrade automatically if configured to do so. To
+best support users, the upgrade to a version of the Compliance Operator with
+this enhancement must do one of two things.
+
+1. Version all supported profiles and make `version` a required property of the `Profile` CRD
+2. Make `version` an optional property, and version profiles incrementally
+
+In either case, it is expected that user can gracefully upgrade to a release of
+the Compliance Operator that includes this enhancement.
+
+Once a profile is versioned, it is expected to remain versioned for its
+life cycle. This expectation matches reality where profiles are version in some
+way from an authoring body (e.g., PCI-DSS 3.2.1 to 4.0.0, or CIS OpenShift
+1.3.0 to 1.4.0).
+
+### Version Skew Strategy
+
+Not applicable to this particular enhancement given the Compliance Operator is
+a single installation within a single cluster.
+
+### Operational Aspects of API Extensions
+
+Operational impact on this change should be minimal considering the profiles
+need to be parsed regardless, and this change is limited to setting one
+additional property on a CRD.
+
+The compliance content shipped with the Compliance Operator by default may
+change, which will still be true with this enhancement. Users should expect to
+use `Profile` CRDs in the same way they have in the past.
+
+#### Failure Modes
+
+A potential failure mode would be if `version` were a required property of the
+`Profile` CRD and the data stream contained a profile that didn't have a
+version. In that case, profile parsing would fail, and not allow the user to
+use the operator at all. This particular failure could be prevented in CI by
+ensuring each profile contains a version, if it is a required property.
+
+#### Support Procedures
+
+In the case above, users supplying their own content must ensure they're
```

Comment:
As in tailored profile or?

Reply:
Even more elaborate than a TailoredProfile, where someone is crafting and building the datastream using the CaC tools and then loading that into the compliance operator.

This isn't a common case today. I have yet to hear of a user doing this in the wild, but I wanted to highlight it regardless.